src/module/system/damage/weapon.ts

```diff
@@ -391,14 +391,32 @@ class WeaponDamagePF2e {
             }
         }
 
-        // Roll notes
+        // Critical specialization effects
+        const critSpecEffect = ((): CritSpecEffect => {
+            if (!weapon.isOfType("weapon")) return [];
+
+            // If an alternate critical specialization effect is available, apply it only if there is also a
+            // qualifying non-alternate
```

Comment:
Why require this?  Is there a reason to prevent a custom crit spec being applied to a weapon that wasn't assigned to an existing weapon group?  I.e., someone wants to make a new weapon that isn't in a group and give it a new kind of crit spec effect too.

The only use I can see of alternate crit spec text is Spike Launcher, which should work fine without the limitation.

Reply:
An `alternate` is a crit spec that requires one already have the critical specialization feature.

Reply:
I see, the RE really has two different functions.  One is grant crit spec for the strike.  The other is to define an alternate crit spec for the strike, but not grant it.  It does only one at a time.

To create a new weapon, one would put a crit spec RE on the weapon that defines its crit spec, with a predicate to match item.id.  Then some other crit spec RE on a feat or feature would grant crit spec to some kind of group (simple, two-handed, dwarf, etc.) that the weapons happens to be in.